auth/opa/rpcauth/rpcauth.go

```diff
@@ -141,27 +133,20 @@ func (g *Authorizer) Eval(ctx context.Context, input *RPCAuthInput) error {
 		// We've failed so let's see if we can help tell the user what might have failed.
 		hints, err = g.policy.DenialHints(ctx, input)
 		if err != nil {
-			errRegister := recorder.RegisterInt64Counter(authzDenialHintErrorCounterName, authzDenialHintErrorCounterDesc)
-			if errRegister != nil {
-				logger.V(1).Error(errRegister, "failed to register "+authzDenialHintErrorCounterName)
-			}
-			errCounter := recorder.AddInt64Counter(ctx, authzDenialHintErrorCounterName, 1, attribute.String("method", input.Method))
+
+			errCounter := recorder.Counter(ctx, authzDenialHintErrorCounter, 1, attribute.String("method", input.Method))
 			if errCounter != nil {
-				logger.V(1).Error(errCounter, "failed to add counter "+authzDenialHintErrorCounterName)
+				logger.V(1).Error(errCounter, "failed to add counter "+authzDenialHintErrorCounter.Name)
 			}
 			// We can't do much here besides log that something went wrong
 			logger.V(1).Error(err, "failed to get hints for authz policy denial", "error", err)
 		}
 	}
 	logger.V(1).Info("authz policy evaluation result", "authorizationResult", result, "input", input, "denialHints", hints)
 	if !result {
-		errRegister := recorder.RegisterInt64Counter(authzDeniedPolicyCounterName, authzDeniedPolicyCounterDesc)
+		errRegister := recorder.Counter(ctx, authzDeniedPolicyCounter, 1, attribute.String("method", input.Method))
 		if errRegister != nil {
-			logger.V(1).Error(errRegister, "failed to register "+authzDeniedPolicyCounterName)
-		}
-		errCounter := recorder.AddInt64Counter(ctx, authzDeniedPolicyCounterName, 1, attribute.String("method", input.Method))
-		if errCounter != nil {
-			logger.V(1).Error(errCounter, "failed to add counter "+authzDeniedPolicyCounterName)
+			logger.V(1).Error(errRegister, "failed to add counter "+authzDeniedPolicyCounter.Name)
```

Comment:
It seems like we're doing this logging a lot.

Should we have a recorder.CounterOrLog helper function that logs the error instead of returning it?

Reply:
I like that idea